src/NBitcoin/Consensus.cs

```diff
@@ -3,133 +3,101 @@
-        /// </summary>
+        /// <inheritdoc />
         public long CoinbaseMaturity { get; set; }
 
-        /// <summary>
-        /// Amount of coins mined when a new network is bootstrapped.
-        /// Set to Money.Zero when there is no premine.
-        /// </summary>
-        public Money PremineReward { get; set; }
-
-        /// <summary>
-        /// The height of the block in which the pre-mined coins should be.
-        /// Set to 0 when there is no premine.
-        /// </summary>
-        public long PremineHeight { get; set; }
-
-        /// <summary>
-        /// The reward that goes to the miner when a block is mined using proof-of-work.
-        /// </summary>
-        public Money ProofOfWorkReward { get; set; }
-
-        /// <summary>
-        /// The reward that goes to the miner when a block is mined using proof-of-stake.
-        /// </summary>
-        public Money ProofOfStakeReward { get; set; }
-
-        /// <summary>
-        /// Maximal length of reorganization that the node is willing to accept, or 0 to disable long reorganization protection.
-        /// </summary>
-        public uint MaxReorgLength { get; set; }
-
-        /// <summary>
-        /// The maximum amount of coins in any transaction.
-        /// </summary>
-        public long MaxMoney { get; set; }
+        /// <inheritdoc />
+        public Money PremineReward { get; }
 
-        public ConsensusOptions Options { get; set; }
+        /// <inheritdoc />
+        public long PremineHeight { get; }
 
-        public class BuriedDeploymentsArray
-        {
-            private readonly int[] heights;
-
-            public BuriedDeploymentsArray()
-            {
-                this.heights = new int[Enum.GetValues(typeof(BuriedDeployments)).Length];
-            }
-
-            public int this[BuriedDeployments index]
-            {
-                get => this.heights[(int)index];
-                set => this.heights[(int)index] = value;
-            }
-        }
+        /// <inheritdoc />
+        public Money ProofOfWorkReward { get; }
 
-        public class BIP9DeploymentsArray
-        {
-            private readonly BIP9DeploymentsParameters[] parameters;
-
-            public BIP9DeploymentsArray()
-            {
-                this.parameters = new BIP9DeploymentsParameters[Enum.GetValues(typeof(BIP9Deployments)).Length];
-            }
-
-            public BIP9DeploymentsParameters this[BIP9Deployments index]
-            {
-                get => this.parameters[(int)index];
-                set => this.parameters[(int)index] = value;
-            }
-        }
+        /// <inheritdoc />
+        public Money ProofOfStakeReward { get; }
+
+        /// <inheritdoc />
+        public uint MaxReorgLength { get; }
+
+        /// <inheritdoc />
+        public long MaxMoney { get; }
+
+        public ConsensusOptions Options { get; set; }
 
-        public Consensus()
+        public Consensus(IConsensus consensus)
```

Comment:
Some of the fields are above the constructor, some are below. 
Can you please move all of them to be before the constructor?